src/app/main/component/auth/components/sign-in/sign-in.component.spec.ts

```diff
@@ -212,16 +213,64 @@ describe('SignIn component', () => {
-        });
-      });
-    }));
+    it('should navigate to ubs-admin/orders for ROLE_UBS_EMPLOYEE', () => {
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_UBS_EMPLOYEE');
+      component.onSignInSuccess(mockData);
+      expect(router.navigate).toHaveBeenCalledWith(['ubs-admin', 'orders']);
+    });
+
+    it('should navigate to profile/userId for ROLE_USER', () => {
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_USER');
+      spyOn(component, 'navigateToPage').and.returnValue(['profile', mockData.userId]);
+      component.onSignInSuccess(mockData);
+      expect(router.navigate).toHaveBeenCalledWith(['profile', mockData.userId]);
+    });
+
+    it('should navigate to events/eventId with params for isEventsDetails', () => {
+      component.isEventsDetails = true;
+      component.eventId = 45;
+      component.isOwnerParams = true;
+      component.isActiveParams = false;
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_OTHER');
+      spyOn(component, 'navigateToPage').and.callThrough();
+      component.onSignInSuccess(mockData);
+      expect(router.navigate).toHaveBeenCalledWith(['/events', 45, { isOwner: true, isActive: false }]);
+    });
+
+    it('should navigate to ubs-admin/orders for ROLE_UBS_EMPLOYEE', () => {
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_UBS_EMPLOYEE');
+      const result = component.navigateToPage({});
+      expect(result).toEqual(['ubs-admin', 'orders']);
+    });
+
+    it('should navigate to profile/userId for ROLE_USER', () => {
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_USER');
+      const result = component.navigateToPage({ userId: 123 });
+      expect(result).toEqual(['profile', 123]);
+    });
+
+    it('should navigate to /events/eventId with params for isEventsDetails', () => {
+      component.isEventsDetails = true;
+      component.eventId = 43;
+      component.isOwnerParams = true;
+      component.isActiveParams = false;
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_OTHER');
+      const userRoleSubjectSpy = spyOn(jwtServiceMock.userRole$, 'next');
+      spyOn(component, 'definitionOfAuthoritiesAndPositions');
+      const result = component.navigateToPage({});
+      expect(result).toEqual(['/events', 43, { isOwner: true, isActive: false }]);
+      expect(userRoleSubjectSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should navigate to the correct page based on user role', () => {
+      spyOn(jwtServiceMock, 'getUserRole').and.returnValue('ROLE_UBS_EMPLOYEE');
+      spyOn(component, 'definitionOfAuthoritiesAndPositions').and.stub();
+      component.isEventsDetails = false;
+      component.isUbs = false;
+      const result = component.navigateToPage({ userId: 'user123' });
+      expect(result).toEqual(['ubs-admin', 'orders']);
+      expect(component.definitionOfAuthoritiesAndPositions).toHaveBeenCalled();
```

Comment:
you can use `(component as any).definitionOfAuthoritiesAndPositions` to test this private method except to do it public

Reply:
fixed all mistakes